src/octree/batch_iterator.rs

```diff
@@ -151,16 +148,27 @@ impl<'a> BatchIterator<'a> {
     where
         F: FnMut(PointData) -> Result<()>,
     {
-        let mut point_stream =
-            PointStream::new(self.batch_size, self.local_from_global.clone(), &mut func);
-        let mut iterator: Box<Iterator<Item = Point>> = match &self.culling {
-            PointCulling::Any() => Box::new(self.octree.all_points()),
-            PointCulling::Aabb(aabb) => Box::new(self.octree.points_in_box(aabb)),
-            PointCulling::Obb(obb) => Box::new(self.octree.points_in_obb(obb)),
-            PointCulling::Frustum(frustum) => Box::new(self.octree.points_in_frustum(frustum)),
-            PointCulling::OrientedBeam(beam) => Box::new(self.octree.points_in_oriented_beam(beam)),
-        };
-        iterator.try_for_each(|point: Point| point_stream.push_point_and_callback(point))?;
-        point_stream.callback()
+        //todo (catevita) mutable function parallelization
```

Comment:
I'm not sure I understand and/or we need this. In the end, the user calls try_for_each_batch. Are you proposing the callbacks should happen on several threads in parallel? I think that's not super useful, since it requires synchronization on the user side again. Better to (as a user) e.g. request points in non-overlapping boxes from several threads.

Reply:
As told offline, this PR only takes care of implementing the trait, the parallelization follows up next